app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -1763,32 +1763,18 @@ class SectionRenderingService(
         }
     }
 
-    val visibleTableRows = table.rows.filter(isVisibleValueRow)
-    val visibleIndexes = visibleTableRows.zipWithIndex.map(_._2)
-
-    val visibleHeaders = table.header.zipWithIndex
-      .filter { case (_, index) =>
-        visibleIndexes.contains(index)
-      }
-
     val headerNumericClasses = table.rows
+      .find(isVisibleValueRow)
       .map(_.values.map(v => if (isNumeric(v)) "govuk-table__header--numeric" else ""))
-      .headOption
       .getOrElse(List())
 
-    val headerClasses: List[String] = visibleHeaders.map {
-      case (header, _) if header.classes.nonEmpty => header.classes.getOrElse("")
-      case (_, index)                             => headerNumericClasses.lift(index).getOrElse("")
+    val headCells = table.header.zipWithIndex.map { case (header, index) =>
```

Comment:
Isn't the headCell is the first cell in the row? Should it also be filtered by visible as in the original implementation?